### ISSUE TITLE
CI: bundle install in step cucumber fails

### ISSUE THREAD

**Issue by olleolleolle** [vcr/vcr#995]
See https://github.com/vcr/vcr/pull/994 for more details

Not much to go on. The Ruby 2.7 complains of "circular deps". 

<details>

```
Fetching gem metadata from https://rubygems.org/...........
Resolving dependencies...
Using rake 13.0.6
Using base64 0.1.1
Using ruby2_keywords 0.0.5
Fetching bigdecimal 3.1.4
Fetching concurrent-ruby 1.2.2
Fetching connection_pool 2.4.1
Fetching minitest 5.20.0
Installing bigdecimal 3.1.4 with native extensions
Installing connection_pool 2.4.1
Using mutex_m 0.1.2
Using public_suffix 5.0.3
Using childprocess 3.0.0
Using contracts 0.17
Using builder 3.2.4
Fetching middleware 0.1.0
Installing minitest 5.20.0
Installing concurrent-ruby 1.2.2
Using thor 1.2.2
Fetching thread_safe 0.3.6
Installing middleware 0.1.0
Fetching cucumber-tag-expressions 2.0.4
Installing thread_safe 0.3.6
Installing cucumber-tag-expressions 2.0.4
Fetching cucumber-cucumber-expressions 10.3.0
Using diff-lcs 1.5.0
Using multi_test 0.1.2
Using ffi 1.16.3
Using rspec-support 3.12.1
Using bundler 2.4.10
Using docile 1.4.0
Using simplecov-html 0.12.3
Using simplecov_json_formatter 0.1.4
Using coderay 1.1.3
Using rexml 3.2.6
Using excon 0.104.0
Using faraday-em_http 1.0.0
Using faraday-em_synchrony 1.0.0
Using faraday-excon 1.1.0
Using faraday-httpclient 1.0.1
Fetching cucumber-html-formatter 6.0.3
Installing cucumber-gherkin 13.0.0
Fetching cucumber-core 7.0.0
Installing cucumber-html-formatter 6.0.3
Installing cucumber-core 7.0.0
Fetching cucumber-wire 3.0.0
Installing cucumber-wire 3.0.0
Fetching cucumber 4.0.0
Installing cucumber 4.0.0
Using aruba 0.14.14
Bundle complete! 22 Gemfile dependencies, 81 gems now installed.
Bundled gems are installed into `./vendor/bundle`
Error: Process completed with exit code 1.
```

</details>

Ruby all of them
Gem current

**Comment by mattbrictson** [comment]
After some investigation ([see this failed job](https://github.com/vcr/vcr/actions/runs/6564654413/job/17831473820?pr=996)), I found that the underlying problem is that protobuf, used internally by cucumber, is incompatible with the recently-released Active Support (Rails) 7.1.

```
bundler: failed to load command: cucumber (/home/runner/work/vcr/vcr/vendor/bundle/ruby/3.2.0/bin/cucumber)
/home/runner/work/vcr/vcr/vendor/bundle/ruby/3.2.0/gems/activesupport-7.1.1/lib/active_support/core_ext/time/conversions.rb:62:in `<class:Time>': undefined method `deprecator' for ActiveSupport:Module (NoMethodError)

  deprecate to_default_s: :to_s, deprecator: ActiveSupport.deprecator
                                                          ^^^^^^^^^^^
Did you mean?  deprecate_constant
[...]
	from gems/protobuf-cucumber-3.10.8/lib/protobuf.rb:10:in `<top (required)>'
```

In Active Support 7.1, gems that use parts of Active Support outside of Rails[^1] must first require the top-level gem:

```ruby
require "active_support"
```

This has to be done prior to requiring other modules within AS.

The protobuf code does not do this. It requires individual modules but never requires the top-level `active_support` itself.

https://github.com/ruby-protobuf/protobuf/blob/b866fc5667226d0582f328ab24c648e578c5a380/lib/protobuf.rb#L7-L11

```ruby
require 'active_support/core_ext/object/blank'
require 'active_support/core_ext/object/try'
require 'active_support/inflector'
require 'active_support/json'
require 'active_support/notifications'
```

VCR could work around this by pinning `gem "active_support", "< 7.1"` in `Gemfile.cucumber`.

And/or, we could push for protobuf to fix the problem, or push Rails to restore the ability to use Active Support modules without requiring the top-level gem, since this change in 7.1 seems to be causing a lot of downstream headaches in legacy gems.

[^1]: https://github.com/rails/rails/issues/49495#issuecomment-1749085658

**Comment by mattbrictson** [comment]
I also just realized that VCR is using an old version of cucumber (~> 4.0). The latest version of cucumber (9.0) no longer depends on protobuf and doesn't have the problem I mentioned.

So upgrading cucumber (which might be more involved) would be a better long-term solution, versus pinning Active Support.
